### PR TITLE
[processor/resourcedetection]: Populate "aws.ecs.task.id" property for ECS Tasks (#8274)

### PR DESCRIPTION
**Pull request by mkielar**
**Description:**
  The `resourcedetection` processor now populates `aws.ecs.task.id` property (in addition to other `aws.ecs.task.*`
  properties). This simplifies configuration of `awsemfexporter`, which internally searches for `aws.ecs.task.id` 
  property when using `TaskId` placeholder in `loggroup` / `logstream` name template.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8274

**Testing:** ECS Task ARNs come in two versions. In the old one, the last part of the ARN contains only the `task/<task-id>`. In the new one, it contains `task/cluster-name/task-id`.  Implementation and Unit Tests have been added to handle both cases.

**Documentation:** `README.md` now also mentions `aws.ecs.task.id` as inferred property for ECS.

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
> Generated code is out of date, please run "make generate" and commit the changes in this PR.

File: processor/resourcedetectionprocessor/internal/aws/ecs/metadata.yaml [hunk 1]
```diff
@@ -35,6 +35,10 @@ resource_attributes:
     description: The aws.ecs.task.family
     type: string
     enabled: true
+  aws.ecs.task.id:
+    description: The aws.ecs.task.id
```
Comment (mx-psi) [inline processor/resourcedetectionprocessor/internal/aws/ecs/metadata.yaml:39]: Please add a description of this. For the other fields, the description is on the semantic conventions repository, but this one has not description anywhere. You may also want to update the processor's README

**Review by mx-psi — COMMENTED** [review]
`aws.ecs.task.id` is not specified in the semantic conventions https://github.com/open-telemetry/semantic-conventions/blob/v1.23.1/docs/resource/cloud-provider/aws/ecs.md. While this is not a hard requirement, it would greatly help in reviewing this PR and ensuring the value on this attributes matches what the `awsefmexporter` and other present or future components expect. Would you be willing to file a PR on semantic-conventions first?

File: processor/resourcedetectionprocessor/internal/aws/ecs/metadata.yaml (hunk 1 above)
Comment (mkielar) [reply processor/resourcedetectionprocessor/internal/aws/ecs/metadata.yaml:39]: Hey @mx-psi,
Thanks for spotting that. I've updated README to mention new attribute, and I've created a new PR: https://github.com/open-telemetry/semantic-conventions/pull/597

**Comment by mx-psi** [comment]
Can you add a changelog note? You can use `make chlog-new` to generate a new one from the template

**Review by mx-psi — APPROVED** [review]
LGTM. Would like someone from AWS (@bryan-aguilar ? @Aneurysm9 ?) to take a look at this before we merge

**Review by bryan-aguilar — APPROVED** [review]
LGTM. Thank you for filing the semantic conventions PR also.

**Pull request merged**